src/unwinder/sentry_unwinder_dbghelp.c

```diff
@@ -43,16 +43,32 @@ sentry__unwind_stack_dbghelp(
     memset(&stack_frame, 0, sizeof(stack_frame));
 
     size_t size = 0;
-#if defined(_WIN64)
+#if defined(_M_X64)
     int machine_type = IMAGE_FILE_MACHINE_AMD64;
     stack_frame.AddrPC.Offset = ctx.Rip;
     stack_frame.AddrFrame.Offset = ctx.Rbp;
     stack_frame.AddrStack.Offset = ctx.Rsp;
-#else
+#elif defined(_M_IX86)
     int machine_type = IMAGE_FILE_MACHINE_I386;
     stack_frame.AddrPC.Offset = ctx.Eip;
     stack_frame.AddrFrame.Offset = ctx.Ebp;
     stack_frame.AddrStack.Offset = ctx.Esp;
+#elif defined(_M_ARM64)
+    int machine_type = IMAGE_FILE_MACHINE_ARM64;
+    stack_frame.AddrPC.Offset = ctx.Pc;
+# if defined (NONAMELESSUNION)
+    stack_frame.AddrFrame.Offset = ctx.DUMMYUNIONNAME.DUMMYSTRUCTNAME.Fp;
+# else
+    stack_frame.AddrFrame.Offset = ctx.Fp;
+# endif
```

Comment:
this looks weird, is that a mingw thing? Why is it only needed for `Fp` and not all the others?

Reply:
I probably should have added a comment on this - this is a Windows header thing (reference: [with some details](https://devblogs.microsoft.com/oldnewthing/20170907-00/?p=96956)). 

I have no particular insight into the why, but that’s how it is for this particular structure (reference: https://learn.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-arm64_nt_context). So this will match the headers, and work regardless of whether `NONAMELESSUNION` is defined.